packages/core/src/input/InputManager.ts

```diff
@@ -0,0 +1,320 @@
+import { Ray, Vector2 } from "@oasis-engine/math";
+import { Engine } from "../Engine";
+import { Entity } from "../Entity";
+import { HitResult } from "../HitResult";
+import { Script } from "../Script";
+import { Input } from "./Input";
+import { Pointer } from "./Pointer";
+
+/**
+ * Input Manager.
+ */
+export class InputManager {
+  private static _tempRay: Ray = new Ray();
+  private static _tempPoint: Vector2 = new Vector2();
+  private static _tempHitResult: HitResult = new HitResult();
+
+  private _engine: Engine;
+  private _canvas: HTMLCanvasElement;
+
+  private _multiTouchEnabled: boolean = false;
+  /** Current simulated input. */
+  private _input: Input = new Input();
+  private _eventList: PointerEvent[] = [];
+  private _activePointerCount: number = 0;
+  private _pointerList: Pointer[] = [];
+  private _pointers: Pointer[] = [];
+  private _pointerIdToIndexMap: Record<number, number> = {};
+
+  /**
+   *	Get pointers.
+   *  @remarks The returned list should be considered deep-read-only.
+   * 	@return Pointers
+   */
+  get pointers(): Readonly<Pointer[]> {
+    return this._pointers;
+  }
+
+  /**
+   *	Get input.
+   *  @remarks The returned list should be considered deep-read-only.
+   * 	@return Input
+   */
+  get input(): Readonly<Input> {
+    return this._input;
+  }
+
+  /**
+   *  Whether to handle multi-touch.
+   */
+  get multiTouchEnabled(): boolean {
+    return this._multiTouchEnabled;
+  }
+
+  set multiTouchEnabled(enabled: boolean) {
+    this._multiTouchEnabled = enabled;
+  }
+
+  /**
+   * Constructor an InputManager.
+   * @param engine - The current engine instance
+   */
+  constructor(engine: Engine) {
+    this._engine = engine;
+    // @ts-ignore
+    const canvas = (this._canvas = engine.canvas._webCanvas);
+    canvas.addEventListener("pointerdown", this._pushEventList);
+    canvas.addEventListener("pointerup", this._pushEventList);
+    canvas.addEventListener("pointermove", this._pushEventList);
+    canvas.addEventListener("pointercancel", this._pushEventList);
+    canvas.addEventListener("pointerout", this._pushEventList);
+  }
+
+  /**
+   * Called when the engine is destroyed.
+   */
+  destroy(): void {
```

Comment:
destroy is internal function, should named '_destroy()' and add '@internal'.

Reply:
Done.